index.bs

```diff
@@ -909,7 +972,49 @@ partial interface HTMLIFrameElement {
       <li>Return "<code>Disabled</code>".</li>
     </ol>
   </section>
+  <section>
+    <h3 id="report-violation">Generate report for violation of |feature| policy
+    with |message| and |disposition|, on |settings|</h3>
+
+    <p> Given a feature (|feature|), a string (|message|), another string
+    (|disposition|), an <a>environment settings object</a> (|settings|), and an
+    optional string (|group|), this algorithm generates a <a>report</a>,
+    containing |message|, about the <a>violation</a> of the policy on
+    |feature|.</p>
+
+    1.  Let |body| be a new {{FeaturePolicyViolationReportBody}}, initialized
+        as follows:
+
+        :   [=FeaturePolicyViolationReportBody/featureId=]
+        ::  |feature|'s string representation.
+        :   [=FeaturePolicyViolationReportBody/message=]
+        ::  |message|
+        :   [=FeaturePolicyViolationReportBody/sourceFile=]
+        ::  null
+        :   [=FeaturePolicyViolationReportBody/lineNumber=]
+        ::  null
+        :   [=FeaturePolicyViolationReportBody/columnNumber=]
+        ::  null
+        :   [=FeaturePolicyViolationReportBody/disposition=]
+        ::  |disposition|
+
+    2.  If the user agent is currently executing script, and can extract the
+        source file's URL, line number, and column number from |settings|, then
+        set |body|'s [=FeaturePolicyViolationReportBody/sourceFile=],
+        [=FeaturePolicyViolationReportBody/lineNumber=], and
+        [=FeaturePolicyViolationReportBody/columnNumber=] accordingly.
+
+    3.  If |group| is omitted, set |group| to "default".
+
+    4.  Execute [[reporting/#queue-report]] with |body|,
+        "feature-policy-violation", |group|, and |settings|.
+
+    Note: This algorithm should be called when a feature policy has
+    been <dfn data-lt="violation|violated">violated</dfn>. It is up to each
```

Comment:
I'd like to put the definition of "violation" in the "Reporting" section, rather than in a note here, if it's possible. We can leave the note, and have it link to that definition.